### PR TITLE
Replaced index by uuid // Added cancel button

### PR DESCRIPTION
**Pull request by felixcanditt**
Thanks for your review :)

Things that changed:

- I replaced index by uuid for a smoother handling of filter/remove functions
- I added a cancel button to the forms in Watchlist and Library that works while adding or editing an item
- Some code rewriting that doesn't change content

### CODE REVIEW

**Review by ThomasKirst — CHANGES_REQUESTED** [review]
Great work overall and good naming of components and functions!

Only a few remarks :)

File: client/src/App.js [hunk 1]
```diff
@@ -21,6 +21,7 @@ export default function App() {
   );
 
   const [itemToBeEdited, setItemToBeEdited] = useState();
+  console.log(itemToBeEdited);
```
[the remark this answers is not in the document]
Comment (felixcanditt) [reply client/src/App.js:24]: removed

File: client/src/App.js [hunk 2]
```diff
@@ -34,49 +35,45 @@ export default function App() {
     setWatchlist([newItem, ...watchlist]);
   }
 
-  function removeFromWatchlist(itemToBeRemoved, indexToBeRemoved) {
+  function editWatchlist(editedItem) {
+    const editedWatchlist = watchlist.filter(
```
[the remark this answers is not in the document]
Comment (felixcanditt) [reply client/src/App.js:39]: renamed

File: client/src/components/WatchlistForm.js [hunk 3]
```diff
@@ -88,9 +96,17 @@ export default function WatchlistForm({
...
-      <WatchlistFormOptions formItem={formItem} onChangeItem={changeItem} />
-
-      <button>speichern</button>
+      <WatchlistFormOptions
+        formItem={formItem}
+        onUpdateFormItem={updateFormItem}
+      />
+
+      <Buttons>
+        <button type="reset" onClick={() => handleFormCancelation()}>
```
[the remark this answers is not in the document]
Comment (felixcanditt) [reply client/src/components/WatchlistForm.js:105]: adjusted

File: client/src/components/LibraryCards.js [hunk 4]
```diff
@@ -11,12 +11,11 @@ export default function LibraryCards({
         <Card key={index}>
```
[the remark this answers is not in the document]
Comment (felixcanditt) [reply client/src/components/LibraryCards.js:11]: adjusted

**Review by anitaschiller — CHANGES_REQUESTED** [review]

File: package.json [hunk 5]
```diff
@@ -6,8 +6,7 @@
   "scripts": {
     "test": "echo \"Error: no test specified\" && exit 1",
```
[the remark this answers is not in the document]
Comment (felixcanditt) [reply package.json:7]: adjusted

File: package.json [hunk 6]
```diff
@@ -6,8 +6,7 @@
...
     "client": "cd ./client && npm start",
-    "server": "cd ./server && npm start",
-    "dev": "concurrently \"npm run server\" \"npm run client\"",
+    "dev": "npm run client",
```
[the remark this answers is not in the document]
Comment (felixcanditt) [reply package.json:9]: adjusted

**Pull request merged**